specs/dotsigstore_bundle/pb/bundle.proto

```diff
@@ -0,0 +1,167 @@
+        bytes signature = 2;
+}
+
+// RekorKindVersion contains the entry's kind and api version.
+message RekorKindVersion {
+        // Kind is the type of entry being stored in the log.
+        // See here for a list: https://github.com/sigstore/rekor/tree/main/pkg/types
+        string kind = 3;
+        // The specific api version of the type.
+        string version = 4;
+}
+
+// InclusionProof is the proof returned from the transparency log. Can
+// be used for on line verification against the log.
+message InclusionProof {
+        // The index of the entry in the log.
+        int64 log_index = 1;
+        // The hash value stored at the root of the merkle tree at the time
+        // the proof was generated, hex encoded.
+        string root_hash = 2;
+        // The size of the merkle tree at the time the proof was generated.
+        int64 tree_size = 3;
+        // A list of hashes required to compute the inclusion proof, sorted
+        // in order from leaf to root. Each hash value is hex encoded.
+        repeated string hashes = 4;
+        // The signed tree head that the inclusion proof is based on. The
+        // signature is base64 encoded.
+        string checkpoint = 5;
+}
+
+// TransparencyLogEntry is a separate type, capturing all the details required
+// from Rekor to reconstruct an entry, given that the payload is provided
+// via other means.
+// This type can easily be created from the existing response from Rekor.
+// Future iterations could rely on Rekor returning the minimal set of
+// attributes (excluding the payload) that are required for verifying the
+// inclusion promise. The inclusion promise (called SignedEntryTimestamp
+// the response from Rekor) is similar to a Signed Certificate Timestamp
+// as described here https://www.rfc-editor.org/rfc/rfc9162#name-signed-certificate-timestam.
+message TransparencyLogEntry {
+        // The index of the entry in the log.
+        int64 log_index = 1;
+        // The unique (binary) identifier of the log, as a hex-encoded string.
+        string log_id = 2;
+        // Even if kind can be derived from the attestation type it's kept
+        // as this struct could be automatically created by the sigstore
+        // library based on the response from Rekor. Also during verification
+        // this struct can be passed to the library for Rekor entry
+        // reconstruction. By duplicating the kind here we would so minimize
+        // any possibility of error due to attestation/kind matching logic,
+        // as that logic must be hard-coded in the client. With this construct
+        // it's only determined during publish to Rekor.
+        RekorKindVersion kind_version = 3;
+        // The UNIX timestamp from the log when the entry was persisted.
+        int64 integrated_time = 4;
+        // The inclusion promise is calculated by Rekor. It's calculated as a
+        // signature over the persisted entry, the log ID, log index and the
```

Comment:
"Signature over..."

That's not quite enough details for me to verify the signature. How are all those things encoded?

Reply:
This is updated, can we resolve?